client/src/partials/admin/transaction_type/transaction_type.js

```diff
@@ -0,0 +1,109 @@
+'use strict';
+
+// controller definition
+angular.module('bhima.controllers')
+.controller('TransactionTypeController', TransactionTypeController);
+
+// dependencies injection
+TransactionTypeController.$inject = [
+  'TransactionTypeService', 'NotifyService', 'ModalService',
+  '$translate'
+];
+
+/** Transaction Type Controller  */
+function TransactionTypeController(TransactionType, Notify, Modal, $translate) {
+  var vm = this;
+
+  // global variables
+  vm.gridOptions = {};
+  vm.gridApi = {};
+
+  // edit button template
+  var editTemplate = '<div class="ui-grid-cell-contents">' +
+    '<a title="{{ \'FORM.LABELS.EDIT\' | translate }}" href="" ' +
+    'ng-click="grid.appScope.editType(row.entity)" ' +
+    'data-edit-type="{{ row.entity.id }}">' +
+    '<i class="fa fa-edit"></i> ' +
+    '</a></div>';
+
+  // text template
+  var textTemplate = '<div class="ui-grid-cell-contents">{{ row.entity.text | translate }}</div>';
+
+  // grid default options
+  vm.gridOptions.appScopeProvider = vm;
+  vm.gridOptions.columnDefs       =
+    [
+      { field : 'id', displayName : 'FORM.LABELS.CODE',
+        headerCellFilter: 'translate', width: 100},
+
+      { field : 'text', displayName : 'FORM.LABELS.TEXT',
+        headerCellFilter: 'translate', cellTemplate: textTemplate},
+
+      { field : 'description', displayName : 'FORM.LABELS.DESCRIPTION',
+        headerCellFilter: 'translate'},
+
+      { field : 'type', displayName : 'FORM.LABELS.TYPE',
+        headerCellFilter: 'translate',
+        cellTemplate: 'partials/templates/grid/transactionType.tmpl.html'},
+
+      { field : 'prefix', displayName : 'FORM.LABELS.PREFIX',
+        headerCellFilter: 'translate'},
+
+      { field : 'action', displayName : '...',
+        width: 25,
+        cellTemplate: editTemplate,
+        enableFiltering: false,
+        enableColumnMenu: false
+      }
+    ];
+
+  // register API
+  vm.gridOptions.onRegisterApi = onRegisterApi;
+
+  // expose to the view
+  vm.addType = addType;
+  vm.editType = editType;
+
+  /** API register function */
+  function onRegisterApi(gridApi) {
+    vm.gridApi = gridApi;
+  }
+
+  // add new transaction type
+  function addType() {
+    var request = { action : 'create' };
+
+    return Modal.openTransactionTypeActions(request)
+    .then(function (res) {
+      if (!res) { return; }
+      startup();
+      Notify.success($translate.instant('FORM.INFO.SAVE_SUCCESS'));
```

Comment:
I think `Notify` automatically translates text for you.  You shouldn't need `$translate.instant()` here.